.github/workflows/apps_wallet_prod_build.yml

```diff
@@ -39,6 +39,48 @@ jobs:
           echo "artifact_name=${artifact_name}" >> $GITHUB_ENV
       - name: Build Wallet
         run: pnpm wallet build
+      - name: Get Previous Tag
+        id: prev_tag
+        run: |
+          tags=$(git tag --list 'wallet-v*.*.*' --sort=-creatordate)
+          current_tag=$(echo "$tags" | sed -n 1p)
+          prev_tag=$(echo "$tags" | sed -n 2p)
+          if [ -z "$prev_tag" ]; then
+            echo "No previous tag found. Skipping changelog generation."
+            echo "PREV_TAG=none" >> $GITHUB_ENV
+          else
+            echo "PREV_TAG=$prev_tag" >> $GITHUB_ENV
+          fi
+          echo "CURRENT_TAG=$current_tag" >> $GITHUB_ENV
+
+      - name: Generate Changelog
+        id: generate_changelog
+        run: |
+          if [ "${{ env.PREV_TAG }}" = "none" ]; then
+            echo "No previous tag found. Skipping changelog generation."
+            echo "changelog=No previous tag found. Changelog generation skipped." >> $GITHUB_ENV
+          else
+            git log ${{ env.PREV_TAG }}..${{ env.CURRENT_TAG }} --pretty=format:"- %s by @%an in #%h" -- ./apps/wallet > CHANGELOG.md
+            changelog=$(cat CHANGELOG.md)
+            echo "changelog=$changelog" >> $GITHUB_ENV
```

Comment:
Out of curiosity, when is GITHUB_ENV used then?

Reply:
`$GITHUB_ENV` is a not a normal env, its how you can move a locally-defined env var to the entire workflow so the next steps can use what you put in there, kinda like outputs

In this case, I am putting the `$changelog` env var

https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions#setting-an-environment-variable